### ISSUE TITLE
Ability to use Nested Child Query with parent child inner hits

### ISSUE THREAD

**Issue by dashford** [ongr-io/ElasticsearchDSL#168]
Hi,

I'm currently using the `HasChildQuery` class to include additional documents within my result set. I need to include the `inner_hits` part of the query in order to get the right results but it looks like this ability isn't part of the `HasChildQuery` implementation when dealing with parent/child documents (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-inner-hits.html#parent-child-inner-hits).

Do you know if this is planned to be included in later versions? I saw that `inner_hits` support was added for other queries but not for parent child ones.

**Comment by saimaz** [comment]
Yes, it is in the plans. `inner_hits` is quite complex, that's why it's not there yet.
